langwatch/src/server/features/trace-processing/library/core/__tests__/eventStream.test.ts

```diff
@@ -0,0 +1,209 @@
+          ordering: (a, b) => b.timestamp - a.timestamp,
+        });
+
+        expect(stream.isEmpty()).toBe(true);
+      });
+    });
+  });
+
+  describe("isEmpty", () => {
+    describe("when stream has no events", () => {
+      it("returns true", () => {
+        const stream = new EventStream("1", []);
+
+        expect(stream.isEmpty()).toBe(true);
+      });
+    });
+
+    describe("when stream has events", () => {
+      it("returns false", () => {
+        const events: Event[] = [
+          { aggregateId: "1", timestamp: 1000, type: "A", data: {} },
+        ];
+
+        const stream = new EventStream("1", events);
+
+        expect(stream.isEmpty()).toBe(false);
+      });
+    });
+  });
+
+  describe("getMetadata", () => {
+    describe("when stream is empty", () => {
+      it("returns null for first timestamp", () => {
+        const stream = new EventStream("1", []);
+
+        expect(stream.getMetadata().firstEventTimestamp).toBeNull();
+      });
+
+      it("returns null for last timestamp", () => {
+        const stream = new EventStream("1", []);
+
+        expect(stream.getMetadata().lastEventTimestamp).toBeNull();
+      });
+    });
+
+    describe("when stream has events", () => {
+      it("returns correct first timestamp", () => {
+        const events: Event[] = [
+          { aggregateId: "1", timestamp: 1000, type: "A", data: {} },
+          { aggregateId: "1", timestamp: 2000, type: "B", data: {} },
+        ];
+
+        const stream = new EventStream("1", events);
+
+        expect(stream.getMetadata().firstEventTimestamp).toBe(1000);
+      });
+
+      it("returns correct last timestamp", () => {
+        const events: Event[] = [
+          { aggregateId: "1", timestamp: 1000, type: "A", data: {} },
+          { aggregateId: "1", timestamp: 2000, type: "B", data: {} },
+        ];
+
+        const stream = new EventStream("1", events);
+
+        expect(stream.getMetadata().lastEventTimestamp).toBe(2000);
+      });
+    });
+
+    describe("when aggregateId is numeric", () => {
+      it("converts to string in metadata", () => {
+        const events: Event<number>[] = [
+          { aggregateId: 123, timestamp: 1000, type: "A", data: {} },
+        ];
+
+        const stream = new EventStream(123, events);
+
+        expect(stream.getMetadata().aggregateId).toBe("123");
+      });
+    });
+
+    describe("when aggregateId is string", () => {
+      it("preserves string in metadata", () => {
+        const events: Event[] = [
+          { aggregateId: "test-123", timestamp: 1000, type: "A", data: {} },
+        ];
+
+        const stream = new EventStream("test-123", events);
+
+        expect(stream.getMetadata().aggregateId).toBe("test-123");
+      });
+    });
+  });
+
+  describe("getAggregateId", () => {
+    it("returns the aggregate id", () => {
+      const stream = new EventStream("test-id", []);
+
+      expect(stream.getAggregateId()).toBe("test-id");
```

Comment:
Why isn't this part of `getMetadata()`? It seems like we have divergent patterns here.